Show abi.encoding/decoding example (#1746)
smartcontractkit/documentation: src/content/chainlink-functions/tutorials/abi-decoding.mdx

     subscriptionId: 139,
     totalCostInJuels: 200304341399547731n,
     responseBytesHexstring: '0x000000000000000000000000000000000000000000000000000003e1b870b6320000000000000000000000000000000000000000000000000000000065c12153000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000009425443202f205553440000000000000000000000000000000000000000000000',
     errorString: '',
     returnDataBytesHexstring: '0x',
     fulfillmentCode: 0
   }

   ✅ Raw response:  0x000000000000000000000000000000000000000000000000000003e1b870b6320000000000000000000000000000000000000000000000000000000065c12153000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000009425443202f205553440000000000000000000000000000000000000000000000

   ✅ Fetched BTC / USD price: 4267996919346 (updatedAt: 1707155795) (decimals: 8) (description: BTC / USD)
   ```

   The output of the example gives you the following information:

>    - Your request is first run on a sandbox environment to ensure it is correctly configured.

Author: @khadni, your suggestion is more concise, but the 1st one is preferable for  a broader audience with varying levels of English proficiency,

   - The fulfillment costs are estimated before making the request.
   - Your request was successfully sent to Chainlink Functions. The transaction in this example is `0x660bc9bd4c85645209295c957d0764823b94a1d351a5ead380113e5352e01fb5` and the request ID is `0xa790aaaca3fdcdd0ca0b0e68bf0c699d84da8d59f8b6dd95aa3d4046a3801c27`.

   - The DON successfully fulfilled your request. The total cost was: `0.20180840394238 LINK`.

>    - The consumer contract received a response in hexadecimal string with a value of `0x000000000000000000000000000000000000000000000000000003e1b870b6320000000000000000000000000000000000000000000000000000000065c12153000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000009425443202f205553440000000000000000000000000000000000000000000000`. This value is the ABI encoded response of the `latestRoundData`, `decimals`, and `description` of the _BTC / USD_ price feed. This value is then decoded and stored in the consumer contract.

Author: From CL Functions' perspective, it is not raw.

- [ethers](https://docs.ethers.org/v5/): Ethers.js library, enables the script to interact with the blockchain.
- `@chainlink/functions-toolkit`: Chainlink Functions NPM package. All its utilities are documented in the [NPM README](https://github.com/smartcontractkit/functions-toolkit/blob/main/README.md).
- `@chainlink/env-enc`: A tool for loading and storing encrypted environment variables. Read the [official documentation](https://www.npmjs.com/package/@chainlink/env-enc) to learn more.
- `../abi/functionsDecoder.json`: The abi of the contract your script will interact with. **Note**: The script was tested with this [FunctionsConsumerDecoder contract](https://remix.ethereum.org/#url=https://docs.chain.link/samples/ChainlinkFunctions/FunctionsConsumerDecoder.sol).

The script has two hardcoded values that you have to change using your own Functions consumer contract and subscription ID:

```javascript
const consumerAddress = "0x5fC6e53646CC53f0C3575fd2c71b5056c4823f5c" // REPLACE this with your Functions consumer address
const subscriptionId = 139 // REPLACE this with your subscription ID
```

> The primary function that the script executes is `makeRequestMumbai`. This function can be broken into five main parts:

Reviewer: ```suggestion
The primary function that the script executes is `makeRequestMumbai`. This function consists of five main parts:
```
